### PR TITLE
enh(NewConversationDialog): refactor and split component

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* `newGroupConversationStore` is redundant -> move functionality to component
* optimize components

## 🖌️ UI Checklist

| 🏚️ Before | 🏡 After |
|------------|----------|
image | image
image | image

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] ⛑️ Tests are included or not possible

### CODE REVIEW

File: src/components/NewConversationDialog/NewConversationDialog.vue [hunk 1]
```diff
@@ -32,84 +32,47 @@
...
-							:aria-label="t('spreed', 'Enter password')"
-							:value.sync="password" />
-					</div>
-					<ListableSettings v-model="listable" />
-				</div>
+				<NewConversationSetupPage v-show="page === 0"
+					ref="setupPage"
+					:new-conversation.sync="newConversation"
+					:password.sync="password"
+					:listable.sync="listable"
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/NewConversationDialog/NewConversationDialog.vue:39]: But we're updating it with `<ListableSettings />`, storing in `<NewConversationDialog />` and using from there. So, should be sync IMO

[the remark this answers is not in the document]
Comment (DorraJaouad) [reply src/components/NewConversationDialog/NewConversationDialog.vue:39]: Aa yes, you are right, I missed the update from `ListableSettings`

File: src/components/NewConversationDialog/NewConversationSetupPage.vue [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+		hasPassword: {
+			get() {
+				return this.newConversation.hasPassword
+			},
+			set(value) {
+				this.updateNewConversation({ hasPassword: value })
+				if (!value) {
+					this.passwordValue = ''
+				}
```
Comment (DorraJaouad) [inline src/components/NewConversationDialog/NewConversationSetupPage.vue:160]: This is causing an error `TypeError: Cannot read properties of undefined (reading 'value')`

**Review by DorraJaouad — COMMENTED** [review]
Almost ready

File: src/components/NewConversationDialog/NewConversationSetupPage.vue (hunk 2 above)
Comment (Antreesy) [reply src/components/NewConversationDialog/NewConversationSetupPage.vue]: It triggered a computed set, which expects to get event object, not the String
Now it directly updates a value in parent

**Review by DorraJaouad — APPROVED** [review]
Very nice work 🦅.
Tested and nothing is blocking.
Code-wise, great improvement.

**Pull request merged**